osquery/tables/system/posix/ssl_cert.cpp

```diff
@@ -0,0 +1,199 @@
+/*
+ *  Copyright (c) 2014-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the BSD-style license found in the
+ *  LICENSE file in the root directory of this source tree. An additional grant
+ *  of patent rights can be found in the PATENTS file in the same directory.
+ *
+ */
+
+#include <stdio.h>
+#include <string.h>
+#include <string>
+
+#include <osquery/tables.h>
+
+#include <openssl/asn1.h>
+#include <openssl/bio.h>
+#include <openssl/bn.h>
+#include <openssl/err.h>
+#include <openssl/pem.h>
+#include <openssl/x509.h>
+
+#include <boost/filesystem/operations.hpp>
+#include <boost/filesystem/path.hpp>
+
+#define SHORT_STR 256
+
+namespace osquery {
+namespace tables {
+
+namespace fs = boost::filesystem;
+
+static void fillRow(Row& r, X509* cert) {
+  char temp[SHORT_STR + 1];
+
+  // set certificate subject information
+  X509_NAME* subject_name = X509_get_subject_name(cert);
+  X509_NAME_get_text_by_NID(subject_name, NID_commonName, temp, SHORT_STR);
```

Comment:
For each of the OpenSSL calls you're making, would you mind checking the return value and setting the row appropriately? For instance here, you might use something like
```
auto ret = X509_NAME_get_text_by_NID(subject_name, NID_commonName, temp, SHORT_STR);
r["issued_common_name"] = ret == -1 ? "-1" : std::string(temp);
```
If you'd prefer, you could also throw a conditional in here to `LOG(INFO)` or `LOG(WARNING)` that the API call failed to lookup the cert, but some indication of failure might be merited if things go wrong.